src/particles/elements/diagnostics/openPMD.cpp

```diff
@@ -318,7 +318,17 @@ namespace detail
 
         // pinned memory copy
         PinnedContainer pinned_pc = pc.make_alike<amrex::PinnedArenaAllocator>();
+
+        // FIXME: Looks like this only copies particles on rank 0...
+        auto const global_np_before = pc.TotalNumberOfParticles(true, false);
+        auto const local_np_before = pc.TotalNumberOfParticles(true, true);
         pinned_pc.copyParticles(pc, true);  // no filtering
```

Comment:
@atmyers hm, that is confusing: it only copies particles on rank 0 (if we never redistribute and operate on a non-OK() container w/o space charge grid)